src/data.js

```diff
@@ -105,39 +105,88 @@ Client.prototype.json = function(resolver) {
 };
 
 /**
- * Ensure root client exists and has the given accessToken.
+ * Ensure static clients exist and remove all clients prefixed 'static/', if not
+ * in the clients given here.
  *
- * Should only be called if the app is configured with a rootAccessToken.
- * Otherwise, app should assume whatever is in the table storage is the
- * root access token, and that appropriate role is attached.
+ * Each client is given by an object:
+ *         {clientId, accessToken, description, scopes}
+ * , where description will be amended with a section explaining that this
+ * client is static and can't be modified at runtime.
  */
-Client.ensureRootClient = function(accessToken) {
-  assert(typeof accessToken === 'string',
-    'Expected accessToken to be a string');
-  // Create client resolving conflicts by overwriting
-  return this.create({
-    clientId:         'root',
-    description:      'Automatically created `root` client with star scopes ' +
-                      'for bootstrapping API access',
-    accessToken:      accessToken,
-    expires:          taskcluster.fromNow('24 hours'),
-    details: {
-      created:        new Date().toJSON(),
-      lastModified:   new Date().toJSON(),
-      lastDateUsed:   new Date().toJSON(),
-      lastRotated:    new Date().toJSON(),
-      deleteOnExpiration: false,
-    },
-    scopes:           ['*'],
-    disabled:         0,
-  }, true);
+Client.syncStaticClients = async function(clients = []) {
+  // Validate input for sanity (we hardly need perfect validation here...)
+  assert(clients instanceof Array, 'Expected clients to be am array');
+  for (const client of clients) {
+    assert(typeof client.clientId === 'string', 'expected clientId to be a string');
+    assert(typeof client.accessToken === 'string', 'expected accessToken to be a string');
+    assert(typeof client.description === 'string', 'expected description to be a string');
+    assert(client.scopes instanceof Array, 'expected scopes to be an array of strings');
+    assert(client.clientId.startsWith('static/'), 'static clients must have clientId = "static/..."');
+    assert(client.scopes.every(s => typeof s === 'string'), 'scopes must be strings');
+  }
+
+  // description suffix to use for all static clients
+  const descriptionSuffix = [
+    '',
+    '---',
```

Comment:
This should have newlines before and after, otherwise it shows up as a markdown heading